teams.js

```diff
@@ -0,0 +1,387 @@
+const os = require('os');
+const {cfg, Connect} = require('sm-utils');
+const startCase = require('lodash.startcase');
+
+/**
+ * @typedef {import('@smpx/notify').TeamsTypes.Section} Section
+ * @typedef {import('@smpx/notify').TeamsTypes.Action} Action
+ * @typedef {import('@smpx/notify').TeamsTypes.MessageCard} MessageCard
+ */
+
+let packageObj;
+function getPackageInfo() {
+	if (packageObj) return packageObj;
+	try {
+		packageObj = require(`${process.cwd()}/package.json`);
+	}
+	catch(err) {
+		packageObj = {name: 'Unknown', version: '0.0.0'};
+	}
+	return packageObj;
+}
+
+/** @type {Console} */
+let logger;
+function getLogger() {
+	if (logger) return logger;
+	try {
+		// Try using oak
+		// @ts-ignore
+		const {Oak} = require('@smpx/oak');
+		logger = new Oak('Slack');
+	}
+	catch (err) {
+		logger = console;
+	}
+	return logger;
+}
+
+/** 
+ * @param {string} channel `<channelName>[.<webhookName=default>]`
+ * @returns {string}
+ */
+function getTeamsWebhookUrl(channel) {
+	if (!channel) return null;
+
+	let webhookName;
+	([channel, webhookName = 'default'] = channel.split('.'));
+
+	const webhookUrl = cfg(`teams.${channel}`);
+
+	if (!webhookUrl) return null;
+	if (typeof webhookUrl === 'string') return webhookUrl;
+	return webhookUrl[webhookName];
+}
+
+/**
+ * @returns {Section[]}
+ */
+function getDefaultSections() {
+	const {name, version} = getPackageInfo();
+	const facts = [
+		{
+			name: 'Hostname',
+			value: os.hostname(),
+		},
+		{
+			name: 'Node Environment',
+			value: cfg.getEnv(),
+		},
+	];
+
+	let footerSuffix = '';
+	if (process.env.name || process.env.pm_id) {
+		footerSuffix = `| ${process.env.name} ${process.env.pm_id || -1}`
+	}
+
+	return [{
+		activityTitle: 'App Info:',
+		activitySubtitle: `${name} v${version} ${footerSuffix} | ${new Date().toLocaleString()}`,
+		facts,
+	}];
+}
+
+class Teams {
+	/**
+	 * Overwrite this function to skip teams message sending in some cnditions
+	 * and log the message instead. By default skips in test environment
+	 */
+	static logCondition() {
+		return cfg.isTest();
+	}
+
+	constructor({text = '', channel = ''} = {}) {
+		this._themeColor = '439FE0'; // blue
+		/** @type {Section[]} */
+		this._sections = [];
+		/** @type {Error[]} */
+		this._errors = [];
+		/** @type {Action[]} */
+		this._actions = [];
+
+		if (channel) this.channel(channel);
+		if (text) this.text(text);
+	}
+
+	/** @param {string} channel */
+	static set defaultChannel(channel) {
+		this._defaultChannel = channel;
+	}
+
+	/** @returns {string} */
+	static get defaultChannel() {
+		return this._defaultChannel || 'default';
+	}
+
+	/** @param {string} channel */
+	channel(channel) {
+		this._channel = channel;
+		return this;
+	}
+
+	/** @param {string} summary */
+	summary(summary) {
```

Comment:
Add summary in Slack class too, which sets the fallback in the first attachement